File: arcgis.py
import math


def co_plot():
    plname = input("Enter plan name: ")
    fil = open("arcGIS.txt", "a")
    pilnum = int(input("Enter number of pillars:"))
    east = float(input("Enter easting: "))
    nort = float(input("Enter northing: "))
    date = input("Enter survey date: ")
    surv = input("Enter surveyor name: ")
    for i in range(0, pilnum):
        print(plname, file=fil, end=",")
        print(surv, file=fil, end=",")
        print(date, file=fil, end=",")
        print("Pillar {}:".format(i))
        pillar = input("Pillar name: ")
        print(pillar, file=fil, end=",")
        dist = float(input("enter distance: "))
        beard = float(input("enter bearing in degrees: "))
        bearm = float(input("enter bearing in minutes: "))
        beardec = beard + (bearm / 60)
        depar = dist * (math.sin(math.radians(beardec)))
        latt = dist * (math.cos(math.radians(beardec)))
        coneast = east + depar
        connort = nort + latt
        print("Easting: {} ".format(coneast))
        print(coneast, file=fil, end=",")
        print("Northing: {}".format(connort))
        print(connort, file=fil)
    fil.close()
    return


runr = 1
while runr != 0:
    co_plot()
    runr = int(input("Enter 0 to stop"))

File: arcgis.java
import java.util.Scanner;
import java.io.FileWriter;
import java.io.IOException;

public class CoPlot {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int runr = 1;
        while (runr != 0) {
            coPlot(input);
            System.out.print("Enter 0 to stop: ");
            runr = input.nextInt();
        }

        input.close();
    }

    public static void coPlot(Scanner input) {
        System.out.print("Enter plan name: ");
        String plname = input.nextLine();

        FileWriter fil;
        try {
            fil = new FileWriter("arcGIS.txt", true);
        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
            return;
        }

        System.out.print("Enter number of pillars: ");
        int pilnum = input.nextInt();
        input.nextLine();  // Consume the newline character left by nextInt()

        System.out.print("Enter easting: ");
        float east = input.nextFloat();
        input.nextLine();  // Consume the newline character left by nextFloat()

        System.out.print("Enter northing: ");
        float nort = input.nextFloat();
        input.nextLine();  // Consume the newline character left by nextFloat()

        System.out.print("Enter survey date: ");
        String date = input.nextLine();

        System.out.print("Enter surveyor name: ");
        String surv = input.nextLine();

        for (int i = 0; i < pilnum; i++) {
            try {
                fil.write(plname + ",");
                fil.write(surv + ",");
                fil.write(date + ",");
                System.out.printf("Pillar %d:\n", i);
                System.out.print("Pillar name: ");
                String pillar = input.nextLine();
                fil.write(pillar + ",");
                System.out.print("Enter distance: ");
                float dist = input.nextFloat();
                input.nextLine();  // Consume the newline character left by nextFloat()
                System.out.print("Enter bearing in degrees: ");
                float beard = input.nextFloat();
                input.nextLine();  // Consume the newline character left by nextFloat()
                System.out.print("Enter bearing in minutes: ");
                float bearm = input.nextFloat();
                input.nextLine();  // Consume the newline character left by nextFloat()
                float beardec = beard + (bearm / 60);
                float depar = dist * (float) Math.sin(Math.toRadians(beardec));
                float latt = dist * (float) Math.cos(Math.toRadians(beardec));
                float coneast = east + depar;
                float connort = nort + latt;
                System.out.printf("Easting: %f\n", coneast);
                fil.write(Float.toString(coneast) + ",");
                System.out.printf("Northing: %f\n", connort);
                fil.write(Float.toString(connort) + "\n");
            } catch (IOException e) {
                System.out.println("Error: " + e.getMessage());
            }
        }

        try {
            fil.close();
        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}